test/post-911-gib-status/actions/post-911-gib-status.unit.spec.js

```diff
@@ -0,0 +1,171 @@
+            facilityName: 'Purdue University',
+            participantId: '11170323',
+            trainingType: 'UNDER_GRAD',
+            termId: null,
+            hourType: null,
+            fullTimeHours: 12,
+            fullTimeCreditHourUnderGrad: null,
+            vacationDayCount: 0,
+            onCampusHours: 12,
+            onlineHours: 0,
+            yellowRibbonAmount: 0,
+            status: 'Approved',
+            amendments: [
+              {
+                onCampusHours: 'Unknown Type: number,null',
+                onlineHours: 'Unknown Type: number,null',
+                yellowRibbonAmount: 'Unknown Type: number,null',
+                type: 'string',
+                changeEffectiveDate: 'Unknown Type: string,null'
+              }
+            ]
+          }
+        ]
+      }
+    }
+  };
+
+  beforeEach(setup);
+  afterEach(teardown);
+
+  it('dispatches GET_ENROLLMENT_DATA_SUCCESS on successful fetch', (done) => {
+    global.fetch.returns(Promise.resolve({
+      headers: { get: () => 'application/json' },
+      ok: true,
+      json: () => Promise.resolve(successResponse)
+    }));
+    const thunk = getEnrollmentData();
+    const dispatch = sinon.spy();
+    thunk(dispatch)
+      .then(() => {
+        const action = dispatch.firstCall.args[0];
+        expect(action.type).to.equal(GET_ENROLLMENT_DATA_SUCCESS);
+        expect(action.data).to.equal(successResponse.data.attributes);
+      }).then(done, done);
+  });
+
+  it('dispatches GET_ENROLLMENT_DATA_FAILURE on unspecified failure', (done) => {
+    global.fetch.returns(Promise.reject(new Error('Unknown error in apiRequest')));
+    const thunk = getEnrollmentData();
+    const dispatch = sinon.spy();
+    thunk(dispatch)
+      .then(() => {
+        const action = dispatch.firstCall.args[0];
+        expect(action.type).to.equal(GET_ENROLLMENT_DATA_FAILURE);
+      }).then(done, done);
+  });
+
+  it('dispatches GET_ENROLLMENT_DATA_FAILURE on unexpected error without code', (done) => {
+    global.fetch.returns(Promise.reject({}));
+    const thunk = getEnrollmentData();
+    const dispatch = sinon.spy();
+    thunk(dispatch)
+      .then(() => {
+        const action = dispatch.firstCall.args[0];
+        expect(action.type).to.equal(GET_ENROLLMENT_DATA_FAILURE);
+      }).then(done, done);
+  });
+
+  it('dispatches GET_ENROLLMENT_DATA_FAILURE on unexpected error code', (done) => {
+    global.fetch.returns(Promise.reject({
+      errors: [
+        { status: '500' }
+      ]
+    }));
+    const thunk = getEnrollmentData();
+    const dispatch = sinon.spy();
+    thunk(dispatch)
+      .then(() => {
+        const action = dispatch.firstCall.args[0];
+        expect(action.type).to.equal(GET_ENROLLMENT_DATA_FAILURE);
+      }).then(done, done);
+  });
+
+  it('dispatches matching error action on known error code', (done) => {
+    global.fetch.returns(Promise.reject({
+      errors: [
+        { status: '503' }
```

Comment:
I think we could pretty easily map this so we test all our error codes return what we expect...but I guess I don't know what value that would provide. I mean, it'd ensure they don't get messed up or deleted accidentally, but...

:man_shrugging: Not necessary, just a thought.

Reply:
Yeah, I deliberated over this for a while. My thought here was that I didn't want to duplicate the exact application logic, just test that the condition tree works as expected. 🤷‍♂️